proto/process/agent.proto

```diff
@@ -684,6 +689,10 @@ message Connection {
 
 	// serialized HTTPAggregations object summarizing all http transactions recorded for this connection, organized by request path
 	bytes httpAggregations = 37;
+
+	// The stored values corresponds to an index into the `Connections.tags` field
+	// On the backend, the first value is the tagIndex from process.tagsV2.Encode()
+	repeated uint32 tags = 44;
```

Comment:
I don't think this is needed.  If you use the tag encoder, the `TagIndex` field on each connection will refer to the big blog of tag data that you will store in `EncodedConnectionsTags`

Reply:
This field is use for two purpose, from system-probe to the agent it's a tag index to an `[]string` (`Connections.tags`)
I reuse this field/array (1st entry only) to set the TagIndex corresponding to the `EncodedConnctionsTags` blob ; The `EncodedConnctionsTags` are built per connections batch ; ref https://github.com/DataDog/datadog-agent/pull/9171/files#diff-4c0b44136f46566597537722c2e61525e6275f531ccb1e42f12b82c847ad2025R333

Reply:
>On the backend, the first value is the tagIndex from process.tagsV2.Encode()

The trouble with setting this in this manner is that it heap allocates a single element slice which adds up when you consider the number of connections that we process on the backend.

The `TagIndex` field on the `Connection` is what we expect to receive.  You're free of course to use it for intra-process communication within the agent but before sending to the backend we would only want the `TagIndex` field set on the `Connection`

Reply:
good point, I added `TagsIdx` field for this purpose

Reply:
good point, I added TagsIdx field for this purpose
I would set to -1 if no tags are available, WDYT ?